webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {format} from 'date-fns';
+import React, {useCallback, useMemo, useRef, useState} from 'react';
+
+import Button from 'components/common/button/button';
+import {useUserSurvey} from 'components/hooks/survey';
+import LinearScaleQuestion from 'components/surveyPost/linearScaleQuestion/linearScaleQuestion';
+import TextQuestion from 'components/surveyPost/textQuestion/textQuestion';
+
+import type {MattermostWindow} from 'types/mattermost-webapp';
+import type {CustomPostTypeComponentProps, SurveyResponse} from 'types/plugin';
+
+import './style.scss';
+
+const QUESTION_COMPONENTS = {
+    linear_scale: LinearScaleQuestion,
+    text: TextQuestion,
+};
+
+function SurveyPost({post}: CustomPostTypeComponentProps) {
+    const [errorMessage, setErrorMessage] = useState<string>();
+    const draftResponse = useRef<SurveyResponse>({
+        responses: {},
+        dateCreated: format(new Date(), 'dd/MM/yyyy'),
+    });
+
+    const {
+        survey,
+        linearScaleQuestionID,
+        surveyExpired,
+        responsesExist,
+        setResponses,
+    } = useUserSurvey(post.props.surveyID);
+
+    const disabled = responsesExist || surveyExpired;
+
+    const submitSurveyHandler = useCallback(async () => {
+        if (!draftResponse.current ||
+            Object.keys(draftResponse.current?.responses).length === 0
+        ) {
+            return;
+        }
+
+        const response = await submitSurveyResponse();
```

Comment:
I was thinking that the logic for submitting the survey could also be part of that hook as well. My goal is to get as much of the logic around managing the survey into the hook and to keep the component to just rendering the results.

If that ends up being too much though, feel free to ignore this one. I know you've got other stuff to work on with this, and I don't want to bog you down by getting obsessed with hooks suddenly 😛

Reply:
Yep, totally fine